### DISCUSSION TITLE
FirebaseLogging requests initiated when only using Messaging component

### DISCUSSION THREAD

**Discussion by swrobel** [invertase/react-native-firebase#5402] [Q&A] [answered]
What do I make of all of these requests that my RN iOS app is making? I followed the [Messaging usage instructions](https://rnfirebase.io/messaging/usage) after the [Getting Started for iOS instructions](https://rnfirebase.io/#3-ios-setup) and did not expect any analytics data to be automatically submitted. This may be an an [upstream bug](https://github.com/firebase/firebase-ios-sdk/issues/5837), but I thought I would check here first.

Packages:
@react-native-firebase/app (12.0.0)
@react-native-firebase/messaging (12.0.0)

Pods:
Firebase/CoreOnly (8.0.0)
Firebase/Messaging (8.0.0)
FirebaseCore (8.0.0)
FirebaseCoreDiagnostics (8.1.0)
FirebaseInstallations (8.1.0)
FirebaseMessaging (8.0.0)

<img width="1617" alt="Screen Shot 2021-06-08 at 2 07 42 PM" src="https://user-images.githubusercontent.com/157270/121262094-70d39300-c868-11eb-86c7-be33e72f83cf.png">

**Comment by mikehardy** [comment, marked as answer]
Interesting. The linked upstream bug appears right on target - this is not something that will be in our control - we're definitely not initiating them ourselves. You could try https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh to get a squeaky clean react-native-firebase demo, and you can tear that script down to just app and messaging modules if you like by simply commenting all the other stuff out but it would probably be more useful to the do same simply using their quickstart for messaging https://github.com/firebase/quickstart-ios/tree/master/messaging

**Reply by swrobel** [reply]
I've opened [an issue](https://github.com/firebase/firebase-ios-sdk/issues/8220) with firebase-ios-sdk

**Reply by mikehardy** [reply]
Of note this is resolved I think based on the upstream issue, by adding this in Info.plist:

```xml
<key>FirebaseDataCollectionDefaultEnabled</key>
<false/>
```
